src/main/java/com/guysfromusa/carsgame/control/MessageDispatcher.java

```diff
@@ -0,0 +1,79 @@
+package com.guysfromusa.carsgame.control;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.groupingBy;
+import static org.apache.commons.lang3.Validate.notNull;
+
+/**
+ * Created by Sebastian Mikucki, 30.04.18
+ */
+@Component
+@Slf4j
+public class MessageDispatcher implements Runnable {
+
+    private final GameQueue gameQueue;
+
+    private final GameEngine gameEngine;
+
+    private final TaskExecutor taskExecutor;
+
+    @Autowired
+    public MessageDispatcher(GameQueue gameQueue, GameEngine gameEngine, TaskExecutor taskExecutor) {
+        this.gameQueue = notNull(gameQueue);
+        this.gameEngine = notNull(gameEngine);
+        this.taskExecutor = notNull(taskExecutor);
+    }
+
+    @PostConstruct
+    public void init() {
+        taskExecutor.execute(this);
+    }
+
+    @Override
+    public void run() {
+        while (true) {
```

Comment:
are we going to use single Blocking queue or queue per game? please add some TODO to everything which is temporary solution with some description how this should look like at the end because others might think this is final solution